docs/dal/index.rst

```diff
@@ -146,10 +146,156 @@ Table Access Protocol
 
     -- `Table Access Protocol <https://www.ivoa.net/documents/TAP/>`_
 
+
+Consider the following example for using TAP and ADQL, retrieving 5
+objects from the GAIA DR3 database, showing their id, position and
+mean G-band magnitude between 19 - 20:
+
 .. doctest-remote-data::
 
+    >>> import pyvo as vo
     >>> tap_service = vo.dal.TAPService("http://dc.g-vo.org/tap")
-    >>> tap_results = tap_service.search("SELECT TOP 10 * FROM ivoa.obscore")
+    >>> ex_query = """
```

Comment:
The "explore" step is missing here, IMO. Where do the user find the name of the table and columns?

Some services have a `description` that can be used here (similar to the SIA service above)

Others (like this one) have examples:

```
>>> for e in tap_service.examples:
>>> ...     print(e['QUERY'])
```

One can find the names of the tables:
```
[t for t in tap_service.tables.keys()]
```

And the columns in a table:
```
tap_service.search('select * from gaia.dr3lite', maxrec=1).table.columns
```

Reply:
Sorry for the late reply and many thanks for the review. I have added your suggestions accordingly into that location.